onnxruntime/core/providers/cpu/nn/batch_norm_helper.h

```diff
@@ -19,13 +19,10 @@ class BatchNormHelper {
                                        const Tensor* var,
                                        bool is_spatial = true) {
     const auto& x_dims = X->Shape().GetDims();
-    if (x_dims.size() < 2) {
-      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
-                             "Invalid input X: The rank of input X must be atleast 2. Got rank: ", x_dims.size());
-    }
 
-    int64_t num_channels = x_dims[1];
-    int num_feature_dims = static_cast<int>(X->Shape().NumDimensions() - 2);  // the first 2 are respectively - N and C
+    int64_t num_channels = x_dims.size() > 1 ? x_dims[1] : 1;
+    // the first 2 are respectively - N and C. If x_dims size < 2, num_channels defaults to 1.
+    int num_feature_dims = X->Shape().NumDimensions() >= 2 ? static_cast<int>(X->Shape().NumDimensions() - 2) : 1;
```

Comment:
X->Shape().NumDimensions() is actually x_dims.size(), right?

The logic here is when nDims is 2, then num_feature_dims = 0, but if nDims is 1, num_feature_dims = 1. I am thinking maybe we should use 0 instead of 1 here.